base/ca/src/main/java/com/netscape/cms/profile/common/CAEnrollProfile.java

```diff
@@ -531,4 +532,113 @@ public void execute(Request request) throws EProfileException, ERejectException
         }
     }
 
+    private Map<String, byte[]> convertP12Password (Request request) throws EProfileException {
+        String method = "CAEnrollProfile: convertP12Password: ";
+        Map<String, byte[]> returnPass = null;
+
+        String p12passwd = request.getExtDataInString("serverSideKeygenP12Passwd");
+
+        org.mozilla.jss.crypto.X509Certificate transCert = null;
+
+        CAEngine engine = CAEngine.getInstance();
+        CertificateAuthority ca = engine.getCA();
+        CAConfig caConfig = ca.getConfigStore();
+        ConnectorsConfig connectorsConfig = caConfig.getConnectorsConfig();
+        ConnectorConfig kraConnectorConfig = connectorsConfig.getConnectorConfig("KRA");
+
+        try {
+            CryptoManager cm = CryptoManager.getInstance();
+            String transportNickname = kraConnectorConfig.getString("transportCertNickname", "KRA Transport Certificate");
+            transCert = cm.findCertByNickname(transportNickname);
+        } catch (Exception e) {
+            logger.debug(method + "'KRA transport certificate' not found in nssdb; need to be manually setup for Server-Side keygen enrollment");
```

Comment:
This is a misconfiguration so I think we should log the error message using `logger.error()`.